scripts/assign.py

```
@@ -168,6 +249,18 @@ def main():
                         return
 
+                # Check if there's already an open PR linked to this issue
+                pr_exists, existing_pr_number = has_open_pr_for_issue(owner, repo, issue_number, headers)
+                if pr_exists:
+                    comment_body = (
+                        f"@{user_login} ⚠️ This issue already has an open pull request "
+                        f"(#{existing_pr_number}) linked to it. "
+                        f"Please look for other available issues to contribute to.\n\n"
+                    )
+                    print(f"Rejecting assignment: issue #{issue_number} already has open PR #{existing_pr_number}")
+                    requests.post(f"{issue_url}/comments", headers=headers, json={"body": comment_body})
```

Ruff (0.15.2)

[error] Probable use of `requests` call without timeout

(S113)